### PR TITLE
DetailsRow & DetailsRowFields: Supply onRenderItemColumn return type instead of any

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The `DetailsRow` & `DetailsRowFields` `onRenderItemColumn` prop can be typed as `React.ReactNode` instead of `any`.

#### Focus areas to test

- Build should pass
- Existing tests should pass
- Change _should be_ OK w.r.t backwards compatibility since we are tightening the return type

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8046)

### CODE REVIEW

File: common/changes/office-ui-fabric-react/keco-onRenderItemColumn-return-type_2019-02-20-00-57.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "DetailsRow & DetailsRowFields: onRenderItemColumn return type as React.ReactNode",
+      "type": "none"
```
Comment (KevinTCoughlin) [inline common/changes/office-ui-fabric-react/keco-onRenderItemColumn-return-type_2019-02-20-00-57.json]: This should be `minor`

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts [hunk 2]
```diff
@@ -108,7 +108,7 @@ export interface IDetailsRowBaseProps extends IBaseProps<IDetailsRow>, IDetailsI
   /**
    * Callback for rendering an item column
    */
-  onRenderItemColumn?: (item?: any, index?: number, column?: IColumn) => any;
+  onRenderItemColumn?: (item?: any, index?: number, column?: IColumn) => React.ReactNode;
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: We could use a lookup type here @JasonGore, what do you think?

Comment (ThomasMichon) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: Pretty certain `React.ReactNode` is the correct type here.

Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: Is it redundant against `IDetailsListProps`? I think a `Pick` might be better here. I tried this locally and it seemed to work:
```tsx
export interface IDetailsRowBaseProps extends Pick<IDetailsListProps, 'onRenderItemColumn'>, IBaseProps<IDetailsRow>, IDetailsItemProps {
  ...
}
```

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: Using `Pick` to keep the instances of `onRenderItemColumn` in sync. Will follow-up separately with the remainder.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ export interface IDetailsItemProps {
   cellStyleProps?: ICellStyleProps;
 }
 
-export interface IDetailsRowBaseProps extends IBaseProps<IDetailsRow>, IDetailsItemProps {
+export interface IDetailsRowBaseProps extends Pick<IDetailsListProps, 'onRenderItemColumn'>, IBaseProps<IDetailsRow>, IDetailsItemProps {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: need @natalieethell 's work to document all these inherited props ^_^

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: Agreed

**Comment by KevinTCoughlin** [comment]
FYI - I've gotten sign-off from OneDrive / SharePoint folks so I've added the AutoMerge label.

**Comment by cliffkoh** [comment]
Oops the automerge config was disabled earlier... I just re-enabled it.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts (hunk 3 above)
Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: I think api-extractor 7 would just generate something like 

```tsx
interface IDetailsRowBaseProps extends Pick<IDetailsListProps, 'onRenderItemColumn'>, IBaseProps<IDetailsRow>, IDetailsItemProps {
  ...
}
```

in its api.ts file. We could, as a start, link to IDetailsListProps, and then maybe in the next iteration just document those props directly in IDetailsRowBaseProps. Not sure how much work that would be yet, but it's a thought!

**Pull request merged**